openmc/deplete/chain.py

```diff
@@ -0,0 +1,786 @@
+    "(n,a)": ["He4"], "(n,2nd)": ["H2"], "(n,na)": ["He4"], "(n,3na)": ["He4"],
+    "(n,n3a)": ["He4"] * 3, "(n,2na)": ["He4"], "(n,np)": ["H1"],
+    "(n,n2a)": ["He4"] * 2, "(n,2n2a)": ["He4"] * 2, "(n,nd)": ["H2"],
+    "(n,nt)": ["H3"], "(n,nHe-3)": ["He3"], "(n,nd2a)": ["H2", "He4"],
+    "(n,nt2a)": ["H3", "He4", "He4"], "(n,2np)": ["H1"], "(n,3np)": ["H1"],
+    "(n,n2p)": ["H1"] * 2, "(n,2a)": ["He4"] * 2, "(n,3a)": ["He4"] * 3,
+    "(n,2p)": ["H1"] * 2, "(n,pa)": ["H1", "He4"],
+    "(n,t2a)": ["H3", "He4", "He4"], "(n,d2a)": ["H2", "He4", "He4"],
+    "(n,pd)": ["H1", "H2"], "(n,pt)": ["H1", "H3"], "(n,da)": ["H2", "He4"]}
+
+
+class Chain(object):
+    """Full representation of a depletion chain.
+
+    A depletion chain can be created by using the :meth:`from_endf` method which
+    requires a list of ENDF incident neutron, decay, and neutron fission product
+    yield sublibrary files. The depletion chain used during a depletion
+    simulation is indicated by either an argument to
+    :class:`openmc.deplete.Operator` or through the
+    ``depletion_chain`` item in the :envvar:`OPENMC_CROSS_SECTIONS`
+    environment variable.
+
+    Attributes
+    ----------
+    nuclides : list of openmc.deplete.Nuclide
+        Nuclides present in the chain.
+    reactions : list of str
+        Reactions that are tracked in the depletion chain
+    nuclide_dict : OrderedDict of str to int
+        Maps a nuclide name to an index in nuclides.
+    fission_yields : None or iterable of dict
+        List of effective fission yields for materials. Each dictionary
+        should be of the form ``{parent: {product: yield}}`` with
+        types ``{str: {str: float}}``, where ``yield`` is the fission product
+        yield for isotope ``parent`` producing isotope ``product``.
+        A single entry indicates yields are constant across all materials.
+        Otherwise, an entry can be added for each material to be burned.
+        Ordering should be identical to how the operator orders reaction
+        rates for burnable materials.
+    """
+
+    def __init__(self):
+        self.nuclides = []
+        self.reactions = []
+        self.nuclide_dict = OrderedDict()
+        self._fission_yields = None
+
+    def __contains__(self, nuclide):
+        return nuclide in self.nuclide_dict
+
+    def __getitem__(self, name):
+        """Get a Nuclide by name."""
+        return self.nuclides[self.nuclide_dict[name]]
+
+    def __len__(self):
+        """Number of nuclides in chain."""
+        return len(self.nuclides)
+
+    @classmethod
+    def from_endf(cls, decay_files, fpy_files, neutron_files):
+        """Create a depletion chain from ENDF files.
+
+        Parameters
+        ----------
+        decay_files : list of str
+            List of ENDF decay sub-library files
+        fpy_files : list of str
+            List of ENDF neutron-induced fission product yield sub-library files
```

Comment:
Thank you for clearly explaining fpy in the comments.  Helps a lot for the new reader.